server/controller/FlightController.ts

```diff
@@ -114,9 +117,10 @@ router.get("/violations", authToken, async (req, res, next) => {
 
 router.get("/self", authToken, async (req, res, next) => {
   try {
+    // @ts-ignore TODO: Type this stronger
     const flights = await service.getAll({
       includeUnchecked: true,
-      userId: req.user.id,
+      userId: req.user?.id,
```

Comment:
Das ist unnötig, bzw. dazu kann es nie kommen.
Der user wurde in authToken defintiv hinzugefügt. Das Feld id kann in der User Tabelle niemals leer sein.
Das trifft auch auf die anderen Checks zu.

Wir bräuchten hier einen Type RequestAuth oder so ähnlich.